#159987736: Implement user profile endpoints

justega247 opened the pull request:
#### What does this PR do?
- This PR is for the Implementation of the user profile API endpoints to get a profile, and to edit a users profile.
#### Description of Task to be completed?
- Write tests to check against the functionality of the endpoints
- Write code to make the tests pass
#### How should this be manually tested?
- Clone the repository
- Run ```npm install``` to install dependencies
- Navigate to the root directory and run ```npm test```
#### What are the relevant pivotal tracker stories?
- #159987736
#### Screenshots (if appropriate)
![image](https://user-images.githubusercontent.com/31142281/45126562-db14f180-b16b-11e8-8ea3-d2e1ebcb8466.png)
![image](https://user-images.githubusercontent.com/31142281/45126591-08619f80-b16c-11e8-9665-4050b4fafdb0.png)

justega247 committed:
feat(userController): implement user profile endpoints

- write tests to confirm the user profile endpoints are functional erver/test/user.spec.js
- write code to implement the functionality of the endpoints in server/controllers/UserController.js
- implement functionality to find a user by their username in server/utils/utilities.js
- add seeders to add users to the database before testing commences in erver/database/seeders/20180905172717-user.js

[Finishes #159987736]

valentineezeh commented on server/controllers/UserController.js:
```diff
@@ -34,17 +34,18 @@ class UserController {
+        bio
+      })
         .then((user) => {
-          const token = jwt.sign({ userId: user.id }, secret, { expiresIn: '24h' });
```
I think spreading the object for a single key: value in an object into a multiple line is not necessary unless we have multiple keys: values.

eshaibu commented on server/controllers/UserController.js:
```diff
@@ -90,6 +91,65 @@ class UserController {
+    const {
+      email: Email, bio: Bio, image: Image, firstname: Firstname, lastname: Lastname
+    } = req.body.user;
+    userFound.update({
```
You can avoid 
```
const {
      email: Email, bio: Bio, image: Image, firstname: Firstname, lastname: Lastname
    } = req.body.user;
    userFound.update({
      email: Email || userFound.email,
      bio: Bio || userFound.bio,
      image: Image || userFound.image,
      firstname: Firstname || userFound.firstname,
      lastname: Lastname || userFound.lastname
    })
```
by just 
`user.update(req.body.user)`

cwizard2011 committed:
feat(userController): implement get all user profile

- write tests to confirm the get all user profile endpoint is functional server/test/profile.spec.js
- write code to implement the functionality of the endpoint in server/controllers/UserController.js
- implement functionality to find all users in a database for a logged in user
- add validation method in server/middleware/AuthValidation.js
- add documentation in server/docs/swagger/profile.yaml

[Finishes #159987746]

eshaibu commented on server/database/models/User.js:
```diff
@@ -40,11 +40,18 @@ module.exports = (sequelize, DataTypes) => {
    */
+  User.hook('beforeCreate', (newUser) => {
+    newUser.password = bcrypt.hashSync(newUser.password, bcrypt.genSaltSync(8));
+  });
```
This affects social login
